x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/common/connectors/connector_selector_with_icon.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiLoadingSpinner,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import React, { useMemo, useEffect, useState, useCallback } from 'react';
+import { useAssistantContext } from '@kbn/elastic-assistant';
+import { ConnectorSelector } from '@kbn/security-solution-connectors';
+import {
+  getActionTypeTitle,
+  getGenAiConfig,
+} from '@kbn/elastic-assistant/impl/connectorland/helpers';
+import type { AIConnector } from './types';
+import { useFilteredActionTypes } from './hooks/use_load_action_types';
+import * as i18n from './translations';
+import { useConnectorSelectorWithIconStyles } from './connector_selector_with_icon.styles';
+
+interface Props {
+  isDisabled?: boolean;
+  selectedConnectorId?: string;
+  connectors: AIConnector[];
+  onConnectorSaved?: () => void;
+  onConnectorSelected: (connector: AIConnector) => void;
+  onRefetchConnectors: () => void;
+}
+
+/**
+ * A compact wrapper of the ConnectorSelector with a Selected Icon
+ */
+export const ConnectorSelectorWithIcon = React.memo<Props>(
+  ({
+    isDisabled = false,
+    selectedConnectorId,
+    connectors,
+    onConnectorSaved,
+    onConnectorSelected,
+    onRefetchConnectors,
+  }) => {
+    const { actionTypeRegistry, http, assistantAvailability, toasts } = useAssistantContext();
```

Comment:
It would be better if the assistant context was not used here. This component should be generic, it could be used for the assistant, or not.